### ISSUE TITLE
Consider pffft-fft library support

### ISSUE THREAD

**Issue by G7495x** [MTG/essentia#581]
https://bitbucket.org/jpommier/pffft

- High performance fft library
- BSD-style license
- ARM64 supported
- ARM-NEON supported

Will greatly benefit android devs...

**Comment by dbogdanov** [comment]
Agreed. Is there performance benchmarks comaring pffft with kissfft? It seems there is a patch for KissFFT to support ARM-NEON: 
https://pmeerw.net/blog/programming/neon3.html
https://portland.source.codeaurora.org/patches/quic/webtech/1602-kissfft-add-neon-optimized-routines.patch

**Comment by G7495x** [comment]
The bitbucket repo has benchmarks posted against FFTW, among others... I'm yet to test the numbers against KISS FFT... But here's what's keeping me a bit skeptic about KISS FFT, it was last updated on 2013... With ARM64 devices growing popular, I'm not sure if the code was optimized for this architecture...

Also, Android x86 and x86_64 support for Essentia coming anytime soon...? {ARM, ARM64, x86 and x86_64 architecture support is critical in my opinion...}

**Comment by dbogdanov** [comment]
For now we have [cross-compilation to ARM](http://essentia.upf.edu/documentation/FAQ.html#cross-compiling-for-android) using ```--cross-compile-android``` flag, but you can change desired architecture modifying the [toolchain](https://developer.android.com/ndk/guides/standalone_toolchain.html) in the [waf script](https://github.com/MTG/essentia/blob/master/wscript#L174).

**Comment by carthach** [comment]
It's (relatively) easy to implement another FFT for Essentia, just follow what has been done for Accelerate and Kiss already. A faster non FFTW/Accelerate FFT would be very welcome!
